src/decrypt.rs

```diff
@@ -8,63 +8,42 @@
 
 use crate::{encryption, get_pad_key_and_iv, xor, EncryptedChunk, Error, Result};
 use bytes::Bytes;
-use itertools::Itertools;
-use rayon::prelude::*;
 use std::io::Cursor;
 use xor_name::XorName;
 
 pub fn decrypt(src_hashes: Vec<XorName>, encrypted_chunks: &[&EncryptedChunk]) -> Result<Bytes> {
-    let num_chunks = encrypted_chunks.len();
-    let cpus = num_cpus::get();
-    let batch_size = usize::max(1, (num_chunks as f64 / cpus as f64).ceil() as usize);
-
-    let raw_chunks: Vec<(usize, Bytes)> = encrypted_chunks
-        .chunks(batch_size)
-        .par_bridge()
```

Comment:
I am not sure we want to get rid of parallelization here. Has that been any issue recently?

Reply:
I am on the get it working 100% correct first, then speed up later, if that makes sense